pulsar-client-cpp/include/pulsar/MessageId.h

```diff
@@ -50,34 +47,44 @@ class MessageId {
     /**
      * Serialize the message id into a binary string for storing
      */
-    virtual void serialize(std::string& result) const;
+    void serialize(std::string& result) const;
```

Comment:
Do we still need to serialize and deserialize methods?
We have already overloaded operator<<

Reply:
We have `operator<<` though not `operator>>` and these serve a different purpose in any case.

Serialize/deserialize are serializing the message id in protobuf format (same as `MessageId.serialize()` in Java) and they are used for the Reader so that you can store the message id somewhere else and get it later to restart the reader on a particular message.